crates/matrix-sdk-sqlite/src/state_store.rs

```diff
@@ -1744,17 +1779,17 @@ impl StateStore for SqliteStateStore {
         &self,
         room_id: &RoomId,
         transaction_id: &TransactionId,
-        wedged: bool,
+        error: Option<QueueWedgeError>,
     ) -> Result<(), Self::Error> {
         let room_id = self.encode_key(keys::SEND_QUEUE, room_id);
 
         // See comment in `save_send_queue_event`.
         let transaction_id = transaction_id.to_string();
-
+        let error_value = error.map(|e| self.serialize_value(&e)).transpose()?;
```

Comment:
uber-nit: since the transpose is not trivial, can you add a newline before and after, so we notice there's something going on here, please? (and it'll make the code breathe :))